File: client.java/src/main/java/dev/resolvedor/client/ping/PingClient.java
package dev.resolvedor.client.ping;

import dev.resolvedor.ServiceGenerator;
import retrofit2.Call;
import retrofit2.Response;

public class PingClient {

    private final ServiceGenerator generator;

    public PingClient(String urlBackupServer) {
        this.generator = new ServiceGenerator(urlBackupServer);
    }

    public void ping() {
        PingService service = generator.createService(PingService.class);
        Call<PingResponse> callSync = service.ping();

        try {
            Response<PingResponse> response = callSync.execute();
            if (response.isSuccessful()) {
                PingResponse pingResponse = response.body();
                System.out.println(pingResponse.getMessage());
            } else {
                System.out.println("Request failed with code: " + response.code());
            }
        } catch (Exception e) {
            System.out.println(e.getMessage());
        }
    }

}

File: client.kotlin/app/src/main/kotlin/dev/resolvedor/client/ping/PingClient.kt
package dev.resolvedor.client.ping

import dev.resolvedor.client.ServiceGenerator

class PingClient(urlBackupServer: String) {
    private val generator: ServiceGenerator = ServiceGenerator(urlBackupServer)

    fun ping() {
        val service = generator.createService(serviceClass = PingService::class.java)
        val call = service.getPing()

        try {
            val response = call.execute()
            if (response.isSuccessful) {
                val pingResponse = response.body()
                print(pingResponse?.message)
            } else {
                println("Request failed with code: " + response.code())
            }
        } catch (e: Exception) {
            println(e.message)
        }
    }
}
